dataverse-webapp/src/main/java/edu/harvard/iq/dataverse/harvest/server/OAIRecordServiceBean.java

```diff
@@ -64,9 +65,11 @@ public void updateOaiRecords(String setName, List<Long> datasetIds, Logger setUp
         for (Long datasetId : datasetIds) {
             setUpdateLogger.fine("processing dataset id=" + datasetId);
 
-            datasetRepository.findById(datasetId)
-                .filter(Dataset::containsReleasedVersion)
-                .ifPresent(dataset -> {
+            Optional<Dataset> datasetWithReleasedVersion = datasetRepository
+                                            .findById(datasetId)
+                                            .filter(Dataset::containsReleasedVersion);
+
+            datasetWithReleasedVersion.ifPresent(dataset -> {
```

Comment:
I meant the code inside ifPresent ;) so
```			
ifPresent(dataset -> {
                    setUpdateLogger.fine("found published dataset.");
                    OAIRecord record = recordMap.remove(dataset.getGlobalIdString());

                    if (record == null) {
                        createOaiRecordForDataset(dataset, setName, setUpdateLogger);
                    } else {
                        updateOaiRecordForDataset(dataset, record, setUpdateLogger);
                    })
```
would be

```
ifPresent(this::someMethod);
```

Reply:
I'm not so sure if this is a good idea. Mainly because we would have to hide modifications of `recordMap` map. This will introduce side effect on passed arguments to method. Maybe if we could name it somewhow that it would be clear enough that such side effect is present. But I can't think of the name that will be understood -`updateOaiRecordForDatasetAndRemoveFromMapOrCreateIfNotPresent`. What do you think?

Reply:
Hmmm, yeah it is kinda difficult. I think it should be left as it is then, it is probably needed to refactor the whole class in order to make things consistent.